documentation/decisions/infra/0009-environment-use-cases.md

```diff
@@ -0,0 +1,43 @@
+# Determining The Use Cases Of Our Environments
+
+- **Status:** Active
+- **Last Modified:** 2024-01-24
+- **Related Issue:** [#1055](https://github.com/HHS/simpler-grants-gov/issues/1055)
+- **Author:** Kai Siren
+- **Deciders:** -Aaron Couch, Alsia Plybeah, James Bursa, Michael Chouinard, Sammy Steiner
+
+## Context and Problem Statement
+
+At present we have two environments: `prod`, and `dev`. This setup is insufficient for a team of our size, so this ADR presents an alternative set of environments. The goal is for these environments meet our needs for years to come.
+
+## Decision Drivers
+
+Not having a dedicated place for spinning up testing environments makes certain kinds of changes impossible to test. You must instead "plan and pray" that the changes apply successfully. This can add hours to days to the amount of time it takes to successfully complete a task. `dev` should function as this environment.
+
+Not having a dedicated environment that's always up to date with `main` (and doesn't have random testing changes) makes it harder for non-engineers to know what to test against. `staging` should function as this environment.
+
+## Desired State
+
+- `prod` - This environment will function identically to its existing behavior, wherein it is always the latest Github release.
+- `staging` - This environment will function the way our current `dev` does, wherein it is always deployed from the `main` branch in Github.
+- `dev` - This is a "wild west" environment to which anyone is allowed to push anything at any time.
+
+## Implications On The `dev` Environment
+
+What does it mean for a `dev` to be the "wild west"? Well, specifically:
+
+- Any given engineer can "reserve" a part of dev that is not currently in use. Reserving dev looks like posting to #topic-engineering and saying _"Hi all, I'm going to be deploying `infra/api/database` this sprint to work on ticket 999"_. This sets the understanding that you will be repeatedly deploying to dev, from your laptop, for the duration of the sprint.
+- Dev should not contain anything of any great importance. If you were to wake up Monday morning to your special dev setup having been accidentally deleted, it should not represent a major setback in your plans.
```

Comment:
Might be worth noting that if we ever do have a usecase where we want to have a dedicated environment for testing something for a long period of time, that should probably be stood up as a new environment. For example, if we wanted to test out some new backend component with a large amount of data over the course of a few months (and prod wasn't an option), we'd probably want a dedicated place to do that.